zospy/analyses/new/mtf/fft_through_focus_mtf.py

```diff
@@ -58,8 +62,65 @@ class FFTThroughFocusMTFSettings:
     use_dashes: bool = Field(default=False, description="Use dashes")
 
 
+@analysis_result
+class FFTThroughFocusMTFData:
+    field_coordinate: UnitField[float | tuple[float, float]]
+    data: ValidatedDataFrame
+
+    def to_dataframe(self) -> DataFrame:
+        """Convert the data to a Pandas DataFrame.
+
+        In addition to the columns from FFTThroughFocusMTFData.data, the returned DataFrame has the following columns:
+
+        - FieldX: The field x coordinate
+        - FieldY: The field y coordinate
+
+        Returns
+        -------
+        DataFrame
+            The data in long format.
+        """
+        df: DataFrame = self.data.copy().reset_index()
+
+        df.insert(0, "FieldX", self.field_coordinate.value[0])
+        df.insert(1, "FieldY", self.field_coordinate.value[0])
+
+        return df
+
+
+class FFTThroughFocusMTFResult(RootModel[list[FFTThroughFocusMTFData]]):
+    def __iter__(self):
+        return iter(self.root)
+
+    def __getitem__(self, item):
+        return self.root[item]
+
+    def to_dataframe(self) -> DataFrame:
+        """Convert the data to a Pandas DataFrame.
+
+        The separate dataframes for each field are combined in a DataFrame in long format.
+        In addition to the columns for each wavelength, the returned DataFrame has the following columns:
+
+        - Direction: The direction of the fan, either 'Tangential' or 'Sagittal'.
+        - Field Number: The field number.
+        - Field: The field coordinate.
+
+        Returns
+        -------
+        DataFrame
+            The data in long format.
+        """
+        dataframes = []
+
+        for fft in self:
+            df = fft.to_dataframe()
+            dataframes.append(df)  # Move the index to a separate column to prevent overlap
+
+        return pd.concat(dataframes, ignore_index=True).reset_index(drop=True)
```

Comment:
You can use a list comprehension here:
```suggestion
        return pd.concat([fft.to_dataframe() for fft in self], ignore_index=True).reset_index(drop=True)
```

And reduce the method to a single line

Reply:
You were correct, I ran the wrong `test_mtf` to test. I have now implemented a pydantic basemodel (de)serialization, now all should work